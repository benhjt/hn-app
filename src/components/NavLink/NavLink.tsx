import React, { ReactNode } from 'react';

import { NavLink as RouterNavLink } from 'react-router-dom';
import cn from 'classnames';

type NavLinkProps = {
  children: ReactNode;
  to: string;
};

const NavLink: React.FC<NavLinkProps> = ({ children, to }) => (
  <RouterNavLink
    className={({ isActive }) =>
      cn(isActive ? 'text-white' : 'text-black', 'no-underline mx-2')
    }
    to={to}
  >
    {children}
  </RouterNavLink>
);

export default NavLink;
