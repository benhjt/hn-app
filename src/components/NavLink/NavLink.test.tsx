import { render } from '@testing-library/react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import NavLink from './NavLink';

describe('NavLink', () => {
  it('renders correctly', () => {
    const { container } = render(<NavLink to="/news/1">Top</NavLink>, {
      wrapper: BrowserRouter,
    });
    expect(container).toMatchSnapshot();
  });

  it('renders correctly when link is selected', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/news/1']}>
        <NavLink to="/news/1">Top</NavLink>
      </MemoryRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
