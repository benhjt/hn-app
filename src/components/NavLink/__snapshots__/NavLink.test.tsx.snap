// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLink renders correctly 1`] = `
<div>
  <a
    class="text-black no-underline mx-2"
    href="/news/1"
  >
    Top
  </a>
</div>
`;

exports[`NavLink renders correctly when link is selected 1`] = `
<div>
  <a
    aria-current="page"
    class="text-white no-underline mx-2"
    href="/news/1"
  >
    Top
  </a>
</div>
`;
